app/models/customer.rb

```diff
@@ -1,6 +1,17 @@
 class Customer < ApplicationRecord
   has_many :invoices
+  has_many :merchants, through: :invoices
+  has_many :transactions, through: :invoices
 
   validates :first_name, presence: true
   validates :last_name, presence: true
+
+  def self.top_five_customers
```

Comment:
was good to work through this with you, looks great to me and it's really clear what is going on here